tiledb/api/c_api/query_aggregate/query_aggregate_api_external_experimental.h

```diff
@@ -0,0 +1,207 @@
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_channel_create_operator_count(
+    tiledb_ctx_t* ctx, tiledb_channel_operator_t** op) TILEDB_NOEXCEPT;
+
+/**
+ * Gets the default channel of the query. The default channel consists of all
+ * the rows the query would return as if executed standalone.
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_query_channel_t* default_channel;
+ * tiledb_query_get_default_channel(ctx, query, &default_channel);
+ * @endcode
+ *
+ * @param ctx The TileDB context
+ * @param query A TileDB query
+ * @param channel The channel handle to be allocated
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_query_get_default_channel(
+    tiledb_ctx_t* ctx,
+    tiledb_query_t* query,
+    tiledb_query_channel_t** channel) TILEDB_NOEXCEPT;
+
+/**
+ * Create a channel operation given an input field and an operator.
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_channel_operation_t* sum_A;
+ * tiledb_channel_create_operation_field(ctx, query, operator, "A",
+ * &sum_A);
+ * @endcode
+ *
+ * @param ctx The TileDB context
+ * @param query A TileDB query
+ * @param op The operator to be applied on an attribute
+ * @param input_field_name The attribute on which the operator will be applied
+ * @param operation The operation handle to be allocated
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_channel_create_operation_field(
+    tiledb_ctx_t* ctx,
+    tiledb_query_t* query,
+    const tiledb_channel_operator_t* op,
+    const char* input_field_name,
+    tiledb_channel_operation_t** operation) TILEDB_NOEXCEPT;
+
+/**
+ * Add an aggregate operation on a query channel.
+ * The result computed by the aggregate operation will be available
+ * via the `output_field_name` buffer.
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_channel_add_aggregate(ctx, default_channel, "sumA", sum_A);
+ * @endcode
+ *
+ * @param ctx The TileDB context
+ * @param query A TileDB query
+ * @param channel The query channel on which to apply the operation
+ * @param output_field_name The attribute to be created that holds the result
+ * @param operation The operation to be applied
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_channel_add_aggregate(
+    tiledb_ctx_t* ctx,
+    tiledb_query_channel_t* channel,
+    const char* output_field_name,
+    tiledb_channel_operation_t* operation) TILEDB_NOEXCEPT;
+
+/**
+ * Frees the resources associated with a TileDB channel operation object.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_channel_operation *op;
+ * tiledb_channel_create_operation_field(..., &op);
+ * tiledb_channel_operation_free(&op);
+ * @endcode
+ *
+ * @param op A TileDB channel operation handle
+ */
+TILEDB_EXPORT capi_return_t
+tiledb_channel_operation_free(tiledb_channel_operation_t** op) TILEDB_NOEXCEPT;
```

Comment:
Our old C API `*_free` functions have no capability to return error messages because they're lacking context arguments. (And worse, also with `void` return.) I'd add context arguments here even though they're solely for error reporting.